posthog/src/main/java/com/posthog/PostHogInterface.kt

```diff
@@ -0,0 +1,52 @@
+package com.posthog
+
+public interface PostHogInterface {
+    public fun <T : PostHogConfig> setup(config: T)
+
+    public fun close()
+
+    public fun capture(
+        event: String,
+        distinctId: String? = null, // TODO: should distinctId be here or just in identify?
```

Comment:
* for the _java_ sdk. Yep, this should have distinctId
* for the _android_ sdk. I'd expect it not to have distinct id, but to read the current distinct id (based on whether identify or reset has been called)

Reply:
So I will keep it since it's an optional parameter, which behaves exactly like that, if it's given, it uses the given value, or it uses the cached value.
Kotlin has no overloads so I guess this is fine.